### DISCUSSION TITLE
Audio Preprocessing

### DISCUSSION THREAD

**Discussion by technosaby** [technosaby/gsoc2022#15] [General]
In the Redhen infra, are the audio stream automatically separated from the streams ? Or we need to write some preprocessing for that [similar to this](https://github.com/ksingla025/Speaker_Dia_RedHen/blob/master/01_pre-processing.py). 

References
1. [GSoc 2015](https://www.redhenlab.org/summer-of-code/rsoc15report)

**Comment by brucearctor** [comment]
I imagine Mark or Francis will be best positioned to answer questions about current state of pipelines ...  I'd tag them in here, but don't know their GitHub IDs, @turnermarkb

**Reply by technosaby** [reply]
I have invited Mark and Francis. I will wait for their reply. For now I am extracting audio from video myself and try to do a baselining. Later if there is a pipeline, we can integrate that.

**Reply by brucearctor** [reply]
makes sense -- if easy enough to do yourself, do, so as to not have to wait.  But, better to build on existing componenents, if they are core and to stay around.

**Reply by technosaby** [reply]
Sure, I have created a task and put it to backlog. I will take it after major tasks are complete.

**Comment by technosaby** [comment]
I have one follow up question about the docker creation. Will the docker extract all audio files from video while it is running or will it run on audio files from a location?
- If the docker runs on audio files from a particular location what should be the location where audio files should be kept (is it my Gallina home directory)?
- If the docker extracts all video files to audio files, it will take a long time and may look bad for a demo ?
 @brucearctor @turnermarkb

**Reply by brucearctor** [reply]
Time/Performance less a concern for a demo/PoC -- esp. assuming we have an idea for how to optimize.

**Reply by brucearctor** [reply]
Location is something better answered by @turnermarkb and/or @francis ...  

Related: @turnermarkb -- do we have docs on architecture and mapping pipelines where things run/how/interrelate, and where the source code exists :-) ... I suspect potentially only minimally, but let's see what we do have.

**Reply by brucearctor** [reply]
Beyond a demo, I would imagine not to run anything from a local directory, but again, let's see what the existing conventions are [ and then we can consider ways to improve ]

**Reply by brucearctor** [reply]
I do see: https://www.redhenlab.org/home/the-cognitive-core-research-topics-in-red-hen/red-hen-data-format ( from a slack message ).  This looks to outline at least some on file/data formatting.
